File: 1º Semestre/Progr Comp 1/aula8/ex4.c
// Dois números n1 e n2 são ditos amigos entre si se a soma dos divisores de n1 (excluindo o próprio n1) é igual a n2, e a soma dos divisores de n2 (excluindo o próprio n2) é igual a n1.
// Exemplo: os números 220 e 284 são amigos:
// os divisores do número 220 são: 1, 2, 4, 5, 10, 11, 20, 22, 44, 55, 110
//       a soma dos divisores 1 + 2 + 4 + 5 + 10 + 11 + 20 + 22 + 44 + 55 + 110 = 284
// os divisores do número 284 são: 1, 2, 4, 71, 142
//       a soma dos divisores 1 + 2 + 4 + 71 + 142 =220
// Desenvolva um programa em C que solicite ao usuário informar dois valores inteiros.
// O programa em C deverá verificar se os dois números são amigos e mostrar na tela a mensagem “Os números são amigos.” ou a mensagem “Os números não são amigos.”.

#include <stdio.h>

int main() {
   int n1, n2, soma1=0, soma2=0;

   printf("Digite n1: "); scanf("%d",&n1);
   printf("Digite n2: "); scanf("%d",&n2);

   for (int x=1; x <= n1;x++) {
      if(n1%x==0 && x!=n1) {
         soma1+=x;
      }
   }

   for (int y=1; y <= n2; y++) {
      if(n2%y==0 && y!=n2) {
         soma2+=y;
      }
   }

   if(soma1==n2 && soma2==n1) {
      printf("Os numeros sao amigos");
   } else {
      printf("Os numeros nao sao amigos");
   }

   return 0;
}

File: 1º Semestre/Introd Comp e Tec/aula2/ex4.py
# Desenvolva um programa em Python que solicite ao usuário digitar dois números inteiros. 
# O programa deverá verificar entre estes dois valores qual é o maior e mostrar na tela o maior valor. 
# Se dois valores iguais forem digitados, o algoritmo deverá mostrar a mensagem dizendo que dois números iguais foram digitados.
# Exemplo: se o usuário informar os números 15 e 43, deverá ser mostrado na tela a mensagem “Maior valor: 43”. 
# Se o usuário informar os números 15 e 15, deverá ser mostrado na tela a mensagem “Dois números iguais foram informados”. 

n1 = int(input("Digite o primeiro número: "))
n2 = int(input("Digite o segundo número: "))

if n1>n2:
    print("Maior valor:",n1)
elif n1==n2:
    print("Dois números iguais foram informados")
else:
    print("Maior valor:",n2)